program/src/processor.rs

```diff
@@ -518,8 +518,6 @@ fn process_harvest_rewards(program_id: &Pubkey, accounts: &[AccountInfo]) -> Pro
         // Update the holder rewards state.
         //
         // Temporarily update `unharvested_rewards` with the eligible rewards.
-        holder_rewards_state.last_accumulated_rewards_per_token =
-            pool_state.accumulated_rewards_per_token;
```

Comment:
I can't comment further down, but this creates another potential attack -- if the pool is out of lamports for distribution, such that `rewards_to_harvest` becomes to `0`, I can crank this over and over again to keep adding to my unharvested rewards amount since my `last_accumulated_rewards_per_token` won't ever update.

I can't remember if that's an actual potential situation, but I think we should tighten this up to be totally safe.

How about keeping these lines here, but only advancing `last_accumulated_rewards_per_token` if `eligible_rewards` is greater than 0? I think we're covered in all cases then.

Reply:
You're right, updated. I'll also have a look at the stake program for any of these vectors as well.

Reply:
Looks like stake program covers this edge case in all harvests.

Reply:
Great, thanks for looking into that